Services/service_response.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file service_reponse.c
+ * @author Haoran Qi, Andrew Rooney
+ * @date 2020-06-06
+ */
+#include "service_response.h"
+
+#include <FreeRTOS.h>
+#include <csp/csp.h>
+
+#include "system.h"
+
+extern Service_Queues_t service_queues; // Implemented by the host platform
+
+/**
+ * @brief
+ * 		Wait on a queue of responses to be sent to other CSP nodes (usually the ground)
+ * @details
+ * 		CSP client server will wake when data is in the queue for downlink (telemetery)
+ * @param void * param
+ * 		Not used
+ * @return
+ * 		should not return
+ */
+void service_response_task(void *param) {
+
+    TC_TM_app_id my_address = SYSTEM_APP_ID;
+    csp_packet_t packet;
+    for (;;) {
+      /* To get conn from the response queue */
+      if(xQueueReceive(service_queues.response_queue, &packet,
+                                      NORMAL_TICKS_TO_WAIT) == pdPASS){
+        ex2_log("%d", packet.data[0]);
+      }
+
```

Comment:
@iBacklight This is the main point of this PR: I'm thinking we recieve a packet in this funcion, create a connection with the destination, and send it down!

Reply:
Great idea! just want to know do you mean create a new connection here? I have reviewed the client and server example from csp and socket, and it seems that even if the conn is closed by server, client can still get reply ack from server. The "close conn" function may just close their own port based on the observation that both client and server need to close once. Thus I'm not sure what will happen if we create a new conn here but we can try first!

Reply:
That's also why I choose to use synchonous method to send conn to response queue in the previous version and it superisely worked 0.0

Reply:
I think you're referring to receiving a ping response from the server? There should be no issue, you can open a bunch of connections in CSP and the server can handle them.
In the previous version you were use an asynchronous method, since there's no guarantee that putting the connection on the queue will wake the response queue before the connection is closed. In theory it could've taken a very long time depending on task priorities and if the queue had other things in it.

Reply:
Maybe @EvanGee would have comments here.